content/docs/atom-guide/dev-guide/materials/lua-material-functor-api.md

```diff
@@ -0,0 +1,373 @@
+    end
+end
+```
+
+## Global Utility functions
+
+These functions are available in the global scope.
+
+* **Error**(`string`): Report an error message to `AZ_Error`.
+* **Warning**(`string`): Report a warning message to `AZ_Warning`.
+* **Print**(`string`): Report a message to `AZ_TracePrintf`.
+
+## Process(context) functions
+
+These functions are available in the `context` object that is passed to the `Process` function.
+
+### **GetMaterialPropertyValue\_**
+
+Each **GetMaterialPropertyValue\_** function takes a `string` property name and returns a value of the appropriate type. You must use the version that matches the data type of the material property. The material property must be listed in [GetMaterialPropertyDependencies](#main-functions).
+
+  * **GetMaterialPropertyValue_bool**(`string`): Returns a `boolean`
+  * **GetMaterialPropertyValue_int**(`string`): Returns a `number`
+  * **GetMaterialPropertyValue_uint**(`string`): Returns a `number`
+  * **GetMaterialPropertyValue_enum**(`string`): Returns a `number`
+  * **GetMaterialPropertyValue_float**(`string`): Returns a `number`
+  * **GetMaterialPropertyValue_Vector2**(`string`): Returns a `Vector2` type. See [Lua Math Library](/docs/user-guide/scripting/lua/math-library/).
+  * **GetMaterialPropertyValue_Vector3**(`string`): Returns a `Vector3` type. See [Lua Math Library](/docs/user-guide/scripting/lua/math-library/).
+  * **GetMaterialPropertyValue_Vector4**(`string`): Returns a `Vector4` type. See [Lua Math Library](/docs/user-guide/scripting/lua/math-library/).
+  * **GetMaterialPropertyValue_Color**(`string`): Returns a `Color` type. See [Lua Math Library](/docs/user-guide/scripting/lua/math-library/).
+  * **GetMaterialPropertyValue_Image**(`string`): Returns a generic pointer `userdata` type if an image is available, or `nil` otherwise.
+
+### **HasMaterialProperty(`string`)**
+
+Returns a boolean whether a property with the given name exists. Note the material property does not have to be listed in [GetMaterialPropertyDependencies](#main-functions) because this functions is just checking availability, not *reading values*.
+
+### **SetShaderConstant\_**
+
+Each **SetShaderConstant\_** function takes a `string` shader input name and value to set. You must use the version that matches the data type of the shader input.
+
+  * **SetShaderConstant_bool**(`string`, `boolean`)
+  * **SetShaderConstant_int**(`string`, `number`)
+  * **SetShaderConstant_uint**(`string`, `number`)
+  * **SetShaderConstant_float**(`string`, `number`)
+  * **SetShaderConstant_Vector2**(`string`, `Vector2`)
+  * **SetShaderConstant_Vector3**(`string`, `Vector3`)
+  * **SetShaderConstant_Vector4**(`string`, `Vector4`)
+  * **SetShaderConstant_Color**(`string`, `Color`)
+  * **SetShaderConstant_Matrix3x3**(`string`, `Matrix3x3`)
+  * **SetShaderConstant_Matrix4x4**(`string`, `Matrix4x4`)
+
+### **SetShaderOptionValue\_**
```

Comment:
There's two **SetShaderOptionValue_** sections with the same content, in **Process(context) functions** and **ShaderItem functions**. Is this intentional?

Reply:
The first option is suitable and clear. 
> "context:SetShaderOptionValue_" and "shaderItem:SetShaderOptionValue_"

Yeah, the right-pane TOC not scrolling with you is a pain. I believe there's an issue for it, just hasn't been addressed.

Reply:
I had already gone with option 2. Otherwise I feel we'd have to update every other function heading to use "context:", "shaderItem:" etc, which is a bit much just for this one case of possible confusion.